File: scala-ssl-disabler/SecurityBypasser.scala
// Just copy-paste this code to wherever, and call
// destroyAllSSLSecurityForTheEntireVMForever method to freely enjoy the wild
// world of invlaid SSL certificates.
object SecurityBypasser {
  import java.security.cert.X509Certificate
  import javax.net.ssl._

  private[this] case class Defaults(sslSocketFactory: SSLSocketFactory, hostnameVerifier: HostnameVerifier)
  private[this] var defaults: Option[Defaults] = None

  // The all-permisive trust manager.
  private[this] object AllTM extends X509TrustManager {
    def getAcceptedIssuers: Array[X509Certificate] = null
    def checkClientTrusted(certs: Array[X509Certificate], authType: String) {}
    def checkServerTrusted(certs: Array[X509Certificate], authType: String) {}
  }

  // The all-permissive hostname verifier.
  private[this] object AllHosts extends HostnameVerifier {
    def verify(urlHostName: String, session: SSLSession) = true
  }

  def destroySSLSecurity(): Unit = {
    // Save the current defaults so that they can later be restored.
    defaults = Some(Defaults(HttpsURLConnection.getDefaultSSLSocketFactory, HttpsURLConnection.getDefaultHostnameVerifier))

    // Create the new, permissive SSL context.
    val trustAllCerts = Array[TrustManager](AllTM)
    val sslContext = SSLContext.getInstance("SSL")
    sslContext.init(null, trustAllCerts, null)

    // Set the connection.
    HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory)
    HttpsURLConnection.setDefaultHostnameVerifier(AllHosts)
  }

  def restoreSllSecurity(): Unit = {
    // If defaults is set, restore the originals.
    defaults.foreach { d =>
      HttpsURLConnection.setDefaultSSLSocketFactory(d.sslSocketFactory)
      HttpsURLConnection.setDefaultHostnameVerifier(d.hostnameVerifier)
    }
  }

  // Executes a single function in a bypassed environment.
  def executeBypassed[T](f: => T): T = {
    destroySSLSecurity()
    var r = f
    restoreSllSecurity()
    r
  }
}

File: java-ssl-disabler/src/javax/net/ssl/disabler/SecurityBypasser.java
package javax.net.ssl.disabler;

import java.security.KeyManagementException;
import java.security.NoSuchAlgorithmException;
import javax.net.ssl.*;

public class SecurityBypasser {
    private static boolean isSet = false;
    private static SSLSocketFactory defaultSSLSocketFactory = null;
    private static HostnameVerifier defaultHostnameVerifier = null;

    public static void destroySSLSecurity() {
        try {
            defaultSSLSocketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
            defaultHostnameVerifier = HttpsURLConnection.getDefaultHostnameVerifier();
            isSet = true;

            final TrustManager[] trustAllCerts = new TrustManager[]{new TrustAllManager()};
            final SSLContext sslContext = SSLContext.getInstance("SSL");
            sslContext.init(null, trustAllCerts, null);
            HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
            HttpsURLConnection.setDefaultHostnameVerifier(new AllHosts());
        } catch (final NoSuchAlgorithmException | KeyManagementException e) {
            e.printStackTrace();
        }
    }

    public static void restoreSSLSecurity() {
        if (isSet) {
            HttpsURLConnection.setDefaultSSLSocketFactory(defaultSSLSocketFactory);
            HttpsURLConnection.setDefaultHostnameVerifier(defaultHostnameVerifier);
        }
    }
}
